### PR TITLE
Feature/ninja build

### PR DESCRIPTION
**Pull request by BorjaOuterelo**
- Allows ninja build
- Allows to change compiler
- Only use coverage with GNU compilers.

### CODE REVIEW

File: cmake/SuperBuild.cmake [hunk 1]
```diff
@@ -30,6 +30,8 @@ if(NOT microcdr_FOUND)
         INSTALL_DIR
             ${PROJECT_BINARY_DIR}/temp_install
         CMAKE_ARGS
+            -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+            -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
```
Comment (julianbermudez) [inline cmake/SuperBuild.cmake]: I like it.

File: ci/linux/CMakeLists.txt [hunk 2]
```diff
@@ -16,13 +16,23 @@ cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
...
+
 include(ExternalProject)
+include(CheckCXXCompilerFlag)
 
 set(_c_flags "-fwrapv -fprofile-arcs -ftest-coverage")
 set(_cxx_flags "-fwrapv -fprofile-arcs -ftest-coverage")
 set(_exe_linker_flags "-fprofile-arcs -ftest-coverage")
 set(_shared_linker_flags "-fprofile-arcs -ftest-coverage")
 
+check_cxx_compiler_flag("-fprofile-abs-path" have_fprofile_abs_path)
```
[the remark this answers is not in the document]
Comment (BorjaOuterelo) [reply ci/linux/CMakeLists.txt]: Added C check.

File: CTestConfig.cmake [hunk 3]
```diff
@@ -18,5 +18,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     set(MEMORYCHECK_COMMAND_OPTIONS "${MEMORYCHECK_COMMAND_OPTIONS} --quiet --tool=memcheck --leak-check=yes --show-reachable=yes --num-callers=50 --xml=yes --xml-file=test_%p_memcheck.xml \"--suppressions=${CMAKE_CURRENT_SOURCE_DIR}/ci/valgrind.supp\"")
 
     # Coverage configuration.
-    find_program(COVERAGE_COMMAND NAMES gcov)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```
[the remark this answers is not in the document]
Comment (BorjaOuterelo) [reply CTestConfig.cmake]: Solved.

File: ci/linux/CMakeLists.txt [hunk 4]
```diff
@@ -16,13 +16,23 @@ cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
 
 project(microxrcedds_client_ci LANGUAGES C CXX)
 
+if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```
[the remark this answers is not in the document]
Comment (BorjaOuterelo) [reply ci/linux/CMakeLists.txt]: Solved.

**Pull request merged**